packages/message-relayer/src/service.ts

```diff
@@ -1,188 +1,150 @@
-          if (block.transactions.length !== 1) {
-            throw new Error(
-              `got an unexpected number of transactions in block: ${block.number}`
-            )
-          }
-
-          const messages = await this.state.messenger.getMessagesByTransaction(
-            block.transactions[0].hash
-          )
-
-          // No messages in this transaction so we can move on to the next one.
-          if (messages.length === 0) {
-            this.state.highestCheckedL2Tx++
-            continue
-          }
-
-          // Make sure that all messages sent within the transaction are finalized. If any messages
-          // are not finalized, then we're going to break the loop which will trigger the sleep and
-          // wait for a few seconds before we check again to see if this transaction is finalized.
-          let isFinalized = true
-          for (const message of messages) {
-            const status = await this.state.messenger.getMessageStatus(message)
-            if (
-              status === MessageStatus.IN_CHALLENGE_PERIOD ||
-              status === MessageStatus.STATE_ROOT_NOT_PUBLISHED
-            ) {
-              isFinalized = false
-            }
-          }
-
-          if (!isFinalized) {
-            this.logger.info(
-              `tx not yet finalized, waiting: ${this.state.highestCheckedL2Tx}`
-            )
-            break
-          } else {
-            this.logger.info(
-              `tx is finalized, relaying: ${this.state.highestCheckedL2Tx}`
-            )
-          }
-
-          // If we got here then all messages in the transaction are finalized. Now we can relay
-          // each message to L1.
-          for (const message of messages) {
-            try {
-              await this.state.messenger.finalizeMessage(message)
-            } catch (err) {
-              if (err.message.includes('message has already been received')) {
-                // It's fine, the message was relayed by someone else
-              } else {
-                throw err
-              }
-            }
-            await this.state.messenger.waitForMessageReceipt(message)
-          }
-
-          // All messages have been relayed so we can move on to the next block.
-          this.state.highestCheckedL2Tx++
-        }
+        await this.state.messenger.finalizeMessage(message)
       } catch (err) {
-        this.logger.error('Caught an unhandled error', {
-          message: err.toString(),
-          stack: err.stack,
-          code: err.code,
-        })
+        if (err.message.includes('message has already been received')) {
+          // It's fine, the message was relayed by someone else
+        } else {
+          throw err
+        }
       }
+      await this.state.messenger.waitForMessageReceipt(message)
     }
+
+    // All messages have been relayed so we can move on to the next block.
+    this.state.highestCheckedL2Tx++
   }
 }
+
+if (require.main === module) {
```

Comment:
Adding this small piece of boilerplate means the service can be run by directly executing this file.